rust/private/clippy.bzl

```diff
@@ -23,6 +24,11 @@ load(
 )
 load("//rust/private:utils.bzl", "determine_output_hash", "find_cc_toolchain", "find_toolchain")
 
+CaptureClippyOutputInfo = provider(
```

Comment:
I see we are a bit inconsistent with placing providers. It makes sense to me to put all providers (including CaptureClippyOutputInfo) into providers.bzl. WDYT? (I don't mean to fix pre-existing problems in this PR)